homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -314,3 +408,87 @@ def native_value(self) -> datetime.datetime | float:
         assert envoy.data.inverters is not None
         inverter = envoy.data.inverters[self._serial_number]
         return self.entity_description.value_fn(inverter)
+
+
+class EnvoyEnchargeEntity(CoordinatorEntity[EnphaseUpdateCoordinator], SensorEntity):
+    """Envoy Encharge sensor entity."""
+
+    entity_description: EnvoyEnchargeSensorEntityDescription
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: EnphaseUpdateCoordinator,
+        description: EnvoyEnchargeSensorEntityDescription,
+        serial_number: str,
+    ) -> None:
+        """Initialize Encharge entity."""
+        self.entity_description = description
+        self._serial_number = serial_number
+        envoy_serial_num = coordinator.envoy.serial_number
+        assert envoy_serial_num is not None
+        self._attr_unique_id = f"{serial_number}_{description.key}"
+        assert coordinator.envoy.data is not None
+        assert coordinator.envoy.data.encharge_inventory is not None
+        encharge = coordinator.envoy.data.encharge_inventory[self._serial_number]
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, serial_number)},
+            manufacturer="Enphase",
+            model="Encharge",
+            name=f"Encharge {serial_number}",
+            sw_version=str(encharge.firmware_version),
+            via_device=(DOMAIN, envoy_serial_num),
+        )
+        super().__init__(coordinator)
+
+    @property
+    def native_value(self) -> int | float | datetime.datetime | None:
+        """Return the state of the sensor."""
+        envoy = self.coordinator.envoy
+        assert envoy.data is not None
+        assert envoy.data.encharge_inventory is not None
+        encharge = envoy.data.encharge_inventory[self._serial_number]
+        return self.entity_description.value_fn(encharge)
+
+
+class EnvoyEnchargePowerEntity(
```

Comment:
Isn't this the exact same class as `EnvoyEnchargeEntity`? With the only difference the description being a different class? Can't we merge them?

Reply:
I can't seem to figure out the right typing to make mypy happy for `return self.entity_description.value_fn(encharge)` when i keep them both in the same class. Pushed up what I have (will fail mypy checks) - any thoughts on how to fix that? I would certainly rather have 1 class instead of 2 that are mostly identical.

Reply:
Ooh, now I get why there were 2 classes to begin with. Would a solution be to maybe make a base class with all the duplicate stuff in it? Means you end up with 3 classes, but less repeating.